### PR TITLE
Use the same HTTP client for contacting the bearer token server and t…

### PR DESCRIPTION
**Pull request by umohnani8**
…he registry

Signed-off-by: Miloslav Trmač <mitr@redhat.com>

### CODE REVIEW

**Comment by mtrmac** [comment]
ACK, but
```
The repository 'http://security.ubuntu.com/ubuntu artful-security Release' does not have a Release file.
```
will probably require updating `.travis.Dockerfile` (and hopefully not much more)

File: .travis.Dockerfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
 RUN apt-get -qq update && \
     apt-get install -y sudo docker.io git make btrfs-tools libdevmapper-dev libgpgme-dev libostree-dev
 
-ADD https://storage.googleapis.com/golang/go1.11.1.linux-amd64.tar.gz /tmp
+ADD https://storage.googleapis.com/golang/go1.11.12.linux-amd64.tar.gz /tmp
 
-RUN tar -C /usr/local -xzf /tmp/go1.11.1.linux-amd64.tar.gz && \
-    rm /tmp/go1.11.1.linux-amd64.tar.gz && \
-    ln -s /usr/local/go/bin/* /usr/local/bin/
```
Comment (mtrmac) [inline .travis.Dockerfile]: (Nitpick absolutely not worth another re-spin: removing this indentation looks unintentional.)

**Comment by umohnani8** [comment]
@mtrmac do I need to vendor in the package travis is failing on?

**Comment by mtrmac** [comment]
I guess so — or freeze c/storage at the version relevant to CRI-O 1.13 not to pull in updates.

**Comment by umohnani8** [comment]
Travis is still not happy :/

**Comment by mtrmac** [comment]
I guess that’s the other commit from https://github.com/containers/image/pull/670/files … but eventually I’d expect `make test-skopeo` to break because the current version of skopeo will expect current versions of c/image APIs; so, the tests would have to freeze skopeo at a corresponding version, as well.

Or, I guess, just patch `make test-skopeo` out, if the result of the backport will be sufficiently tested in CRI-O.  I don’t expect any other users to use this branch, and this branch is completely new, so if we got the backport wrong (which is in itself pretty unlikely), we can go back and fix this branch without too much trouble.

**Comment by umohnani8** [comment]
@mtrmac tests are green now. I remove `make test-skopeo` from the .travis.yml

**Pull request merged**